src/components/sdk/MeetingControls/VideoInputBackgroundBlurControl.tsx

```diff
@@ -0,0 +1,172 @@
+  backgroundBlurOffLabel = 'Enable Background Blur',
+}) => {
+  const meetingManager = useMeetingManager();
+  const { devices, selectedDevice } = useVideoInputs(videoInputConfig);
+  const { isVideoEnabled, toggleVideo } = useLocalVideo();
+  const { isBackgroundBlurSupported, createBackgroundBlurDevice } =
+    useBackgroundBlur();
+  const [isVideoTransformCheckBoxOn, setisVideoTransformCheckBoxOn] =
+    useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [
+    dropdownWithVideoTransformOptions,
+    setDropdownWithVideoTransformOptions,
+  ] = useState<ReactNode[] | null>(null);
+  const [activeVideoDevice, setDevice] = useState<
+    Device | VideoTransformDevice | null
+  >(meetingManager.selectedVideoInputDevice);
+
+  // TODO: Move this to the Video Input Provider and expose only one selected Video Input device state
+  useEffect(() => {
+    meetingManager.subscribeToSelectedVideoInputTransformDevice(setDevice);
+    return () => {
+      meetingManager.unsubscribeFromSelectedVideoInputTranformDevice(setDevice);
+    };
+  }, []);
+
+  useEffect(() => {
+    const deviceOptions: ReactNode[] = devices.map((option) => (
+      <PopOverItem
+        key={option.deviceId}
+        children={<span>{option.label}</span>}
+        checked={isOptionActive(selectedDevice, option.deviceId)}
+        onClick={async (): Promise<void> => {
+          // If background blur is on, then re-use the same video transform pipeline, but replace the inner device
+          // If background blur is not on, then do a normal video selection
+          if (isVideoTransformDevice(activeVideoDevice) && !isLoading) {
+            setIsLoading(true);
+            const receivedDevice = videoInputSelectionToDevice(option.deviceId);
+            if (activeVideoDevice instanceof DefaultVideoTransformDevice) {
+              const transformedDevice =
+                activeVideoDevice.chooseNewInnerDevice(receivedDevice);
+              await meetingManager.selectVideoInputDevice(transformedDevice);
+            } else {
+              console.error('Transform device cannot choose new inner device');
+            }
+            setIsLoading(false);
+          } else {
+            await meetingManager.selectVideoInputDevice(option.deviceId);
+          }
+        }}
+      />
+    ));
+    if (isBackgroundBlurSupported) {
+      const videoTransformOptions: ReactNode = (
+        <PopOverItem
+          key="videoinput"
+          children={
+            <>
+              {isLoading && <Spinner width="1.5rem" height="1.5rem" />}
+              {isVideoTransformDevice(activeVideoDevice)
+                ? backgroundBlurOnLabel
+                : backgroundBlurOffLabel}
+            </>
+          }
+          checked={isVideoTransformDevice(activeVideoDevice)}
+          disabled={isLoading}
+          onClick={() => {
+            setIsLoading(true);
+            setisVideoTransformCheckBoxOn((current) => !current);
```

Comment:
Should be be setting the check to `ON` when actually the background blur is applied successfully? It may happen that the video precessing for some reason fails, but the user sees the check still on here.

Reply:
Ill rename to setIsVideoTransformCheckOn